### PR TITLE
Top user profile navigation for small screens - migrate to React

### PR DESCRIPTION
**Pull request by mrkvon**
#### Proposed Changes

* Migrate top profile navigation for small screens to React.

top-navigation-small

#### Testing Instructions

* Go to somebody's profile
* Make the browser smaller
* Check that the top profile navigation hasn't changed in looks and behaviour.
  - especially check that the correct links are displayed and that they behave as they did before
    - own profile: Edit your profile
    - not a contact: Send a message | Add contact
    - some of my contacts: Send a message | Remove contact
    - a contact request I sent: Send a message | Delete contact request
    - a contact request I received: Send a message | Delete contact request
    - additionally, when not own profile && references are enabled: Write a reference

Work on #1019

### CODE REVIEW

File: modules/contacts/client/components/RemoveContact.js [hunk 1]
```diff
@@ -32,10 +40,10 @@ export function RemoveContact({ t, contact, show, inProgress, onHide, onRemoveCo
   const labels = situationLabels[situation];
 
   return (
-    <Modal show={show} onHide={onHide}>
+    <Modal show={show} onHide={onCancel}>
       <div className="modal-content">
         <Modal.Header>
-          <button type="button" className="close" aria-hidden="true" onClick={onHide} ng-if="!removeContactModal.isLoading">&times;</button>
+          <button type="button" className="close" aria-hidden="true" onClick={onCancel} ng-if="!removeContactModal.isLoading">&times;</button>
```
Comment (simison) [inline modules/contacts/client/components/RemoveContact.js]: Remember to remove `ng-if` Angular bits. :-)

Comment (mrkvon) [reply modules/contacts/client/components/RemoveContact.js]: Thank you @simison! :+1:
Fixed in a98e3403d544775fdae6d8e06074382cdabc9a46 as part of #1121.

File: modules/users/client/controllers/profile.client.controller.js [hunk 2]
```diff
@@ -23,11 +23,22 @@ function ProfileController(
   vm.contact = contact;
   vm.contacts = contacts;
 
-  // Exposed to the view
-  vm.hasConnectedAdditionalSocialAccounts = hasConnectedAdditionalSocialAccounts;
-  vm.isConnectedSocialAccount = isConnectedSocialAccount;
-  vm.socialAccountLink = socialAccountLink;
-  vm.isWarmshowersId = isWarmshowersId;
```
Comment (mrkvon) [inline modules/users/client/controllers/profile.client.controller.js]: These are unrelated to the migration itself. Unused variables hanging around.

**Comment by mrkvon** [comment]
I'll be grateful if we can get this old PR finished. I've rebased and fixed some bugs. Would be great if somebody can review and/or approve so we can merge. Thanks!

File: modules/users/client/components/TopNavigationSmall.component.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  username: PropTypes.string.isRequired,
+  selfId: PropTypes.string.isRequired,
+  userId: PropTypes.string.isRequired,
+  contact: PropTypes.object,
+  referencesEnabled: PropTypes.bool.isRequired,
+  isResolved: PropTypes.bool.isRequired,
+  onContactRemoved: PropTypes.func.isRequired,
+};
+
+function LabelWithTooltip({ id, label, tooltip }) {
```
Comment (simison) [inline modules/users/client/components/TopNavigationSmall.component.js]: I think it's fine to leave tooltips out from mobile navigation; they never get shown since there's no _hover_ on mobile devices.

image

That'll simplify this a bit.

Comment (mrkvon) [reply modules/users/client/components/TopNavigationSmall.component.js]: Is the small screen unlikely to be used on desktop?
Also, the tooltip is present in the original code. i.e. this change will degrade/simplify the original.
@simison If you're happy with the above points, I'll proceed with the change. I would appreciate a confirmation.

Comment (simison) [reply modules/users/client/components/TopNavigationSmall.component.js]: I think we can consider everything under 768px as a touch screen, and very unlikely used on a desktop.

If it was on original code it's my bad. 😅 Let's clean out now that we have a chance while looking into it.

Comment (mrkvon) [reply modules/users/client/components/TopNavigationSmall.component.js]: Removed the tooltip in ad97da390e46b965c2a789f0d030eb87dd361388.

File: modules/users/client/components/TopNavigationSmall.component.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+export default function TopNavigationSmall({
+  username,
+  contact,
+  referencesEnabled,
+  isResolved,
+  selfId,
+  userId,
+  onContactRemoved,
+}) {
+  // @TODO this hacky fix should be removed and the data should be consistent
```
Comment (simison) [inline modules/users/client/components/TopNavigationSmall.component.js]: This could warrant opening an issue as a follow up for some day.

**Comment by mrkvon** [comment]
Thank you, @simison for the collaboration!

File: modules/users/client/components/TopNavigationSmall.component.js (hunk 4 above)
Comment (mrkvon) [reply modules/users/client/components/TopNavigationSmall.component.js]: Maybe a reminder to review all those migration-dependent TODOs... seems there will be more as we progress.

I love Typescript, because it helps so much with fixing inconsistencies like this one...
